pkg/fsnotify/fsnotify.go

```diff
@@ -123,23 +161,84 @@ func (w *Watcher) Close() error {
 
 func (w *Watcher) run() {
 	defer w.finished.Done()
+	ticks := time.NewTicker(time.Second)
+	ticksRunning := true
+	deletedPaths := make([]string, 0, 16)
+
 	for {
 		select {
 		case <-w.done:
 			return
 		case event := <-w.private.w.Events:
```

Comment:
Both `w.private.w.Events` and `w.private.w.Errors` are not locked in this loop, though they both should be channels and  probably they aren't reassigned after construction. Usually a `private` sub structure should be locked, so it caught my eye, maybe we should add a comment here.

Reply:
Ok - removed from private :)